src/apis/file/file.service.js

```diff
@@ -9,39 +9,70 @@ class FileService {
     }
 
     createThumb(file, callback) {
-        const image = sharp(file.buffer);
-        image
-            .metadata()
-            .then((metadata, err) => {
-                if (err) {
-                    log.error('Error while converting file ', err);
-                } else {
-                    return image
-                        .resize(Math.round(metadata.width / 2), Math.round(metadata.height / 2))
-                        .jpeg({
-                            quality: 80,
-                            chromaSubsampling: '4:4:4'
-                        })
-                        .toBuffer({ resolveWithObject: true })
-                        .then(({ data, info }) => {
-                            log.info('Converted file info: ', info);
-                            callback(data);
-                        })
-                        .catch(err => {
-                            log.error('err ', err);
-                        });
-                }
+            const image = sharp(file.buffer);
+            image
+                .metadata()
+                .then((metadata, err) => {
+                    if (err) {
+                        log.error('Error while converting file ', err);
+                    } else {
+                        return image
+                            .resize(Math.round(metadata.width / 2), Math.round(metadata.height / 2))
+                            .jpeg({
+                                quality: 80,
+                                chromaSubsampling: '4:4:4'
+                            })
+                            .toBuffer({ resolveWithObject: true })
+                            .then(({ data, info }) => {
+                                log.info('Converted file info: ', info);
+                                callback(data);
+                            })
+                            .catch(err => {
+                                log.error('err ', err);
```

Comment:
Include which service/file the error is arising at while logging errors. Its easier to track errors that way.

Reply:
will do it